google_symptoms/delphi_google_symptoms/date_utils.py

```diff
@@ -0,0 +1,122 @@
+"""utility functions for date parsing."""
+
+from datetime import date, datetime, timedelta
+from itertools import product
+from typing import Dict, List, Tuple
+
+import covidcast
+from delphi_utils.validator.utils import lag_converter
+from pandas import to_datetime
+
+from .constants import COMBINED_METRIC, FULL_BKFILL_START_DATE, PAD_DAYS, SMOOTHERS
+
+
+def generate_patch_dates(params) -> List[Dict[date, List[date]]]:
+    """
+    Generate date range for chunking backfilled dates.
+
+    Parameters
+    ----------
+    params
+
+    Returns
+    -------
+    list of dicts
+    """
+    issue_date = datetime.strptime(params["patch"]["start_issue"], "%Y-%m-%d")
+    end_date = datetime.strptime(params["patch"]["end_issue"], "%Y-%m-%d")
+
+    patch_dates_list = []
+
+    max_expected_lag = lag_converter(params["validation"]["common"].get("max_expected_lag", {"all": 4}))
+    global_max_expected_lag = max(list(max_expected_lag.values()))
+    num_export_days = params["validation"]["common"].get("span_length", 14) + global_max_expected_lag
+
+    while issue_date <= end_date:
+        # start_date gets padded again when executed in run_module
+        expected_start_dt = issue_date - timedelta(days=num_export_days - PAD_DAYS + 2)
+        daterange = generate_query_dates(expected_start_dt, issue_date, num_export_days)
```

Comment:
I understand what this and [this](https://github.com/cmu-delphi/covidcast-indicators/pull/1999#discussion_r1688944510) are doing now, but I find it confusing for various reasons. This is worth revisiting and lightly refactoring so that the code is easier to read. I'll start on that sometime and tag you for review, @aysim319.